lib/msf/core/modules/loader/base.rb

```diff
@@ -457,24 +458,21 @@ def module_path(parent_path, type, module_reference_name)
   #                   the path is not hidden (starts with '.')
   # @return [false] otherwise
   def module_path?(path)
-    !(path.starts_with?(".") ||
-      !path.ends_with?(MODULE_EXTENSION) ||
-      path.match?(UNIT_TEST_REGEX))
+    return false if path.starts_with?(".")
+    return false unless path.ends_with?(MODULE_EXTENSION)
+    return false if path.match?(UNIT_TEST_REGEX)
+    return false unless File.file?(path)
+    return false if script_path?(path)
+
+    true
   end
 
   # Tries to determine if a file might be executable,
   def script_path?(path)
-    # warn users if their external modules aren't marked executable
-    # per #14281
-    if File.directory?(path) || !['#!', '//'].include?(File.read(path, 2))
-      false
-    elsif File.executable?(path)
-      true
-    else
-      # prefer elog since load_error clutters the UI on potential false positives
-      elog("Unable to load module #{path} - LoadError Possible non-executable external module.")
```

Comment:
We'll have to find a better home for this, but I think this makes sense to me in the mean time